source/Irrlicht/COSOperator.cpp

```diff
@@ -102,11 +119,36 @@ void COSOperator::copyToClipboard(const c8 *text) const
 }
 
 
+//! copies text to the primary selection
+void COSOperator::copyToPrimarySelection(const c8 *text) const
+{
+	if (strlen(text)==0)
+		return;
+
+#if defined(_IRR_COMPILE_WITH_SDL_DEVICE_)
+#if SDL_VERSION_ATLEAST(2, 25, 0)
+	if (sdl_supports_primary_selection)
+		SDL_SetPrimarySelectionText(text);
+#endif
+
+#elif defined(_IRR_COMPILE_WITH_X11_DEVICE_)
+    if ( IrrDeviceLinux )
+        IrrDeviceLinux->copyToPrimarySelection(text);
+#endif
+}
+
+
 //! gets text from the clipboard
-//! \return Returns 0 if no string is in there, otherwise an utf-8 string.
 const c8* COSOperator::getTextFromClipboard() const
 {
-#if defined(_IRR_WINDOWS_API_)
+#if defined(_IRR_COMPILE_WITH_SDL_DEVICE_)
+	static char *text = nullptr;

```

Comment:
static variables are bad, put it into the class

Reply:
See https://github.com/minetest/irrlicht/pull/180.